cluster/log-dump.sh

```diff
@@ -25,67 +25,72 @@ KUBE_ROOT=$(dirname "${BASH_SOURCE}")/..
 : ${KUBE_CONFIG_FILE:="config-test.sh"}
 
 source "${KUBE_ROOT}/cluster/kube-util.sh"
+detect-project &> /dev/null
 
 readonly report_dir="${1:-_artifacts}"
 echo "Dumping master and node logs to ${report_dir}"
 
-# Attempts to SSH to a node ($1) and run a simple command. Returns 0 on
-# success and 1 on error.
-function test-ssh() {
-    local -r node_name="${1}"
-    return $(ssh-to-node "${node_name}" "echo test > /dev/null" &> /dev/null)
-}
+# Copy all files /var/log/{$3}.log on node $1 into local dir $2.
+# $3 should be a space-separated string of files.
+# This function shouldn't ever trigger errexit, but doesn't block stderr.
+function copy-logs-from-node() {
+    local -r node="${1}"
+    local -r dir="${2}"
+    local files=(${3})
+    # Append ".log"
+    files=("${files[@]/%/.log}")
+    # Prepend "/var/log/"
+    files=("${files[@]/#/\/var\/log\/}")
+    # Replace spaces with commas, surround with braces
+    local -r scp_files="{$(echo ${files[*]} | tr ' ' ',')}"
 
-# Saves a single output of running a given command ($2) on a given node ($1)
-# into a given local file ($3). Does not fail if the ssh command fails for any
-# reason, just prints an error to stderr.
-function save-log() {
-    local -r node_name="${1}"
-    local -r cmd="${2}"
-    local -r output_file="${3}"
-    if ! ssh-to-node "${node_name}" "${cmd}" > "${output_file}"; then
-        echo "${cmd} failed for ${node_name}" >&2
+    if [[ "${KUBERNETES_PROVIDER}" == "aws" ]]; then
+        local ip=$(get_instance_public_ip "${node}")
+        scp -i "${AWS_SSH_KEY}" "${SSH_USER}@${ip}:${scp_files}" "${dir}" > /dev/null || true
+    else
```

Comment:
We already check provider before calling this function. It might be a problem if people try to use this function elsewhere, but that'd be pretty awkward since it runs the stuff when you source it...